custom_components/wyzeapi/cover.py

```diff
@@ -0,0 +1,156 @@
+    _attr_supported_features = CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE
+    _attr_has_entity_name = True
+
+
+    def __init__(self, camera_service: CameraService, camera: Camera):
+        """Initialize a Wyze garage door."""
+        self._camera = camera
+        if self._camera.type not in [
+            DeviceTypes.CAMERA
+        ]:
+            raise HomeAssistantError(f"Invalid device type: {self._camera.type}")
+        
+        self._camera_service = camera_service
+        self._available = self._camera.available
+        
+    @property
+    def device_info(self):
+        """Return device information about this entity."""
+        return {
+            "identifiers": {(DOMAIN, self._camera.mac)},
+            "name": f"{self._camera.nickname}",
+            "connections": {
+                (
+                    dr.CONNECTION_NETWORK_MAC,
+                    self._camera.mac
+                )
+            }
+        }
+    @property
+    def extra_state_attributes(self):
+        """Return device attributes of the entity."""
+        return {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            "device model": f"{self._camera.product_model}.{self._camera.device_params['dongle_product_model']}"
+        }
+    
+    @property
+    def should_poll(self) -> bool:
+        return False
+    
+    @token_exception_handler
+    async def async_open_cover(self, **kwargs):
+        """Open the cover."""
+        try:
+            await self._camera_service.garage_door_open(self._camera)
+        except (AccessTokenError, ParameterError, UnknownApiError) as err:
+            raise HomeAssistantError(f"Wyze returned an error: {err.args}") from err
+        except Exception as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._camera.garage = True
+            self.async_write_ha_state()
+    
+    @token_exception_handler
+    async def async_close_cover(self, **kwargs):
+        """Close the cover."""
+        try:
+            await self._camera_service.garage_door_close(self._camera)
+        except (AccessTokenError, ParameterError, UnknownApiError) as err:
+            raise HomeAssistantError(f"Wyze returned an error: {err.args}") from err
+        except Exception as err:
+            raise HomeAssistantError(err) from err
+        else:
+            self._camera.garage = False
+            self.async_write_ha_state()
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed."""
+        return self._camera.garage == False
+        
+    @property
+    def available(self):
+        """Return the connection status of this cover."""
+        return self._camera.available
+                
+    @property
+    def unique_id(self):
+        """Define a unique id for this entity."""
+        return f"{self._camera.mac}_GarageDoor"
+
+    @property
+    def name(self):
+        """Return the name of the garage door."""
+        return f"Garage Door"
```

Comment:
Will we have an issue if there are multiple garage doors? Maybe we should append the last four digits of the MAC or something.

Reply:
Home Assistant does some ✨_sparkly magic_✨ and prepends the camera (device) name before this. I don't know why or how, but we shouldn't have any issues here.

Reply:
I've never quite figured out the magic behind that either but it does seem to work 😏